src/applications/disability-benefits/686c-674/config/address-schema.js

```diff
@@ -0,0 +1,244 @@
+chapterKey is used uniquely inside the 686c to check for workflow requirements on individual fields.
+schemaKey is the corresponding schema property name for address. 
+*/
+export const addressUISchema = (schemaKey, chapterKey) => ({
+  'view:livesOnMilitaryBase': {
+    'ui:title':
+      'I live on a United States military base outside of the United States',
+  },
+  'view:livesOnMilitaryBaseInfo': {
+    'ui:description': militaryBaseInfo,
+  },
+  countryName: {
+    'ui:required': formData => isChapterFieldRequired(formData, chapterKey),
+    'ui:title': 'Country',
+    'ui:options': {
+      updateSchema: (formData, schema, uiSchema) => {
+        const countryUI = uiSchema;
+        const countryFormData = formData[`${schemaKey}`];
+        if (formData[`${schemaKey}`]['view:livesOnMilitaryBase']) {
+          countryUI['ui:disabled'] = true;
+          countryFormData.countryName = USA.name;
+          return {
+            enum: [USA.name],
+            default: USA.name,
+          };
+        }
+        countryUI['ui:disabled'] = false;
+        return {
+          type: 'string',
+          enum: countries.map(country => country.label),
+        };
+      },
+    },
+  },
+  addressLine1: {
+    'ui:required': formData => isChapterFieldRequired(formData, chapterKey),
+    'ui:title': 'Street address',
+    'ui:errorMessages': {
+      required: 'Street address is required',
+      pattern: 'Street address must be under 100 characters',
+    },
+  },
+  addressLine2: {
+    'ui:title': 'Street address',
+  },
+  addressLine3: {
+    'ui:title': 'Street address',
+  },
+  city: {
+    'ui:required': formData => isChapterFieldRequired(formData, chapterKey),
+    'ui:errorMessages': {
+      required: 'City is required',
+      pattern: 'City must be under 100 characters',
+    },
+    'ui:options': {
+      replaceSchema: formData => {
+        if (formData[`${schemaKey}`]['view:livesOnMilitaryBase'] === true) {
+          return {
+            type: 'string',
+            title: 'APO/FPO/DPO',
+            enum: militaryCities,
+          };
+        }
+        return {
+          type: 'string',
+          title: 'City',
+          minLength: 1,
+          maxLength: 100,
+          pattern: '^.*\\S.*',
+        };
+      },
+    },
+  },
+  stateCode: {
+    'ui:required': formData =>
+      formData[`${schemaKey}`].countryName === USA.name ||
+      formData[`${schemaKey}`]['view:livesOnMilitaryBase'],
+    'ui:title': 'State',
+    'ui:errorMessages': {
+      required: 'State is required',
+    },
+    'ui:options': {
+      hideIf: formData => {
+        // Because we have to update countryName manually in formData above,
+        // We have to check this when a user selects a non-US country and then selects
+        // the military base checkbox.
+        if (formData[`${schemaKey}`]['view:livesOnMilitaryBase']) {
+          return false;
+        }
+        return formData[`${schemaKey}`].countryName !== USA.name;
+      },
```

Comment:
Relates to the comment above about countryName. Because that value isn't updated when a user selects the military base checkbox, an additional check was needed to actually show/hide the `state` field. The specific case where this is necessary is if a user selects a country other than the US from the country dropdown _and then_ goes back and selects the military base checkbox, `state` will remain hidden. So, we just check the value of `view:livesOnMilitaryBase` as well.